tests/unit/test_estimator.py

```diff
@@ -2219,72 +2226,75 @@ def test_fit_verify_job_name(strftime, sagemaker_session):
     assert fw.latest_training_job.name == JOB_NAME
 
 
-def test_prepare_for_training_with_sagemaker_config_injection_for_partial_debug_hook_config_provided_from_direct_input(
-    sagemaker_session,
-):
-    """
-    Tests that the default_bucket from sagemaker_config is injected into the debugger_hook_config.s3_output_path
-    when only partial debugger_hook_config is provided from direct input
-    """
-    sagemaker_session.sagemaker_config = SAGEMAKER_CONFIG_TRAINING_JOB
-    fw = DummyFramework(
-        entry_point=SCRIPT_PATH,
-        role=ROLE,
-        sagemaker_session=sagemaker_session,
-        instance_count=INSTANCE_COUNT,
-        instance_type=INSTANCE_TYPE,
-        debugger_hook_config=HOOK_CONFIG,
-    )
-    fw._prepare_for_training()
-    assert fw.debugger_hook_config.hook_parameters == HOOK_CONFIG.hook_parameters
-    assert fw.debugger_hook_config.s3_output_path == f"s3://{sagemaker_session.default_bucket()}/"
-
-
-def test_prepare_for_training_with_sagemaker_config_injection_for_full_debug_hook_config_provided_from_direct_input(
+@pytest.mark.parametrize(
+    "debugger_hook_config_direct_input, sagemaker_config, expected_debugger_hook_config_output",
+    [
+        (None, None, S3_OUTPUT_PATH_FROM_SESSION_S3_DEFAULT_CONFIG),
+        (True, None, S3_OUTPUT_PATH_FROM_SESSION_S3_DEFAULT_CONFIG),
+        (False, None, False),
```

Comment:
Optional: For the expected value, would we want to compare against the final value passed to the training job API (rather than the value the constructor is set with)? That way we could make sure the behavior from initialization of the object to where its actually used in the end?